tests/conftest.py

```diff
@@ -1372,3 +1383,27 @@ def local_module_package_index(modules_v2_dir: str) -> Iterator[str]:
             ModuleTool().build(path=path, output_dir=artifact_dir)
         dir2pi(argv=["dir2pi", artifact_dir])
         yield os.path.join(artifact_dir, "simple")
+
+
+@pytest.fixture
+async def migrate_db_from(
```

Comment:
While I don't oppose to this approach, would a factory fixture not be simpler? 

e.g. return a class with a method to perform the actual setup?

Reply:
That was my initial idea but then how would you separate the three stages: setup (restore db dump), migrate (function call in this case), teardown (stop server)? Unless I'm missing something pytest wouldn't be able to handle this anymore so it would become the responsibility of the caller: call the factory, set up, migrate all in a try block, then finally teardown. It makes the test cases itself a bit more bloated, but admittedly simpler in the sense that this marker isn't required.
I must confess this is a domain I'm not that familiar with, I can't confidently say I prefer one way or the other. Considering the above, which way do you lean?

Reply:
I guess this is fine.